File: Frontend/src/types/contracts/history.ts
import { z } from "zod";

export const historyMetricToneSchema = z.enum(["positive", "neutral", "caution", "critical"]);

export const historyStatusSchema = z.enum(["completed", "processing", "abandoned"]);

export const historyMetricSchema = z.object({
  key: z.string(),
  label: z.string(),
  value: z.string(),
  tone: historyMetricToneSchema,
});

export const historySessionSchema = z.object({
  id: z.string(),
  code: z.string(),
  company: z.string(),
  role: z.string(),
  mode: z.string(),
  startedAt: z.string(),
  durationMinutes: z.number(),
  score: z.number(),
  status: historyStatusSchema,
  reportId: z.string().nullable(),
  metrics: z.array(historyMetricSchema),
});

File: Backend/app/schemas/history.py
from app.core.serialization import CamelModel
from app.core.timeutils import UtcDatetime
from app.schemas.common import HistoryStatus, MetricTone


class HistoryMetric(CamelModel):
    """One evaluated dimension of a past session. `value` is already display-ready
    ("High", "94%", "Stable") because the scale differs per metric — the frontend maps
    `key` to an icon and `tone` to a colour, and never reformats the value."""

    key: str
    label: str
    value: str
    tone: MetricTone


class HistorySession(CamelModel):
    id: str
    code: str
    company: str
    role: str
    mode: str
    started_at: UtcDatetime
    duration_minutes: int
    score: int
    status: HistoryStatus
    # null while a session is still `processing` — there is no report to open yet.
    report_id: str | None = None
    metrics: list[HistoryMetric]
